Microsoft.HealthVault.Fhir/Transformers/ObservationToHealthVault.cs

```diff
@@ -32,39 +36,97 @@ public static class ObservationToHealthVault
         /// <returns>The HealthVault thing</returns>
         public static ThingBase ToHealthVault(this Observation observation)
         {
-            // ToDo: detect the type from the codeable values
-            return observation.ToWeight();
+            return observation.ToHealthVault(VocabToHealthVaultHelper.DetectHealthVaultTypeFromObservation(observation));
         }
 
-        private static Weight ToWeight(this Observation observation)
+        private static ThingBase ToHealthVault(this Observation observation, Type type)
         {
-            var weight = new Weight();
-            var value = observation.Value as Quantity;
+            if (type == typeof(Weight))
+            {
+                return observation.ToWeight();
+            }
+
+            if (type == typeof(BloodGlucose))
+            {
+                return observation.ToBloodGlucose();
+            }
+
+            return null;
```

Comment:
No, if we are converting a bundle we would like to skip over for types we do not understand. 

---
In reply to: [124713759](https://github.com/Microsoft/healthvault-fhir-library/pull/3#discussion_r124713759) [](ancestors = 124713759)